python/src/robyn/data/validation/calibration_input_validation.py

```diff
@@ -91,101 +41,237 @@ def check_obj_weights(self, objective_weights: List[float], refresh: bool) -> Va
+                    f"which is considered low confidence."
+                )
 
         return ValidationResult(
-            status=len(error_details) == 0,
-            error_details=error_details,
-            error_message="\n".join(error_messages)
+            status=len(error_details) == 0, error_details=error_details, error_message="\n".join(error_messages)
         )
 
-    def _check_metric_values(self) -> ValidationResult:
+    def _check_lift_values(self) -> ValidationResult:
+        """Check if lift values are valid numbers."""
+        error_details: Dict[Tuple[str, ...], str] = {}
+        error_messages: List[str] = []
+
+        for channel_key, data in self.calibration_input.channel_data.items():
+            if not isinstance(data.lift_abs, (int, float)) or pd.isna(data.lift_abs):
+                error_details[channel_key] = f"Invalid lift value: {data.lift_abs}"
+                error_messages.append(f"Lift value for {'+'.join(channel_key)} must be a valid number.")
+
+        return ValidationResult(
+            status=len(error_details) == 0, error_details=error_details, error_message="\n".join(error_messages)
+        )
+
+    def validate(self) -> List[ValidationResult]:
+        """
+        Implement the abstract validate method from the Validation base class.
+        Returns a list containing the calibration validation result.
+        """
+        return [self.check_calibration()]
+
+    def check_calibration(self) -> ValidationResult:
+        """Check all calibration inputs for consistency and correctness."""
+        if self.calibration_input is None:
+            return ValidationResult(status=True, error_details={}, error_message="")
+
         error_details = {}
         error_messages = []
 
-        dep_var = self.mmmdata.mmmdata_spec.dep_var
+        checks = [
+            self._check_date_range(),
+            self._check_spend_values(),
+            self._check_metric_values(),
+            self._check_confidence_values(),
+            self._check_lift_values(),
+        ]
 
-        for channel, data in self.calibration_input.channel_data.items():
-            if data.metric != DependentVarType(dep_var):
-                error_details[channel] = f"Metric mismatch: {data.metric} vs. {dep_var}"
-                error_messages.append(f"Metric for {channel} does not match the dependent variable ({dep_var}).")
+        for result in checks:
+            if not result.status:
+                error_details.update(result.error_details)
+                error_messages.append(result.error_message)
 
         return ValidationResult(
-            status=len(error_details) == 0,
-            error_details=error_details,
-            error_message="\n".join(error_messages)
+            status=len(error_details) == 0, error_details=error_details, error_message="\n".join(error_messages)
         )
 
-    def validate(self) -> ValidationResult:
+    @staticmethod
+    def create_modified_calibration_input(
```

Comment:
- util function to help test data modification and to help simplify test cases. 
- handle special test cases like non-existent channels